stdlib/public/core/AutoDiff.swift

```diff
@@ -183,6 +183,41 @@ public extension Differentiable where TangentVector == Self {
   }
 }
 
+/// A type that consists of a differentiable vector space and some other
+/// non-differentiable component.
+///
+/// Mathematically, this represents a product manifold that consists of
+/// a differentiable vector space and some arbitrary manifold, where the tangent
+/// bundle of the entire product manifold is equal to the vector space
+/// component.
+///
+/// This abstraction is useful for representing common differentiable data
+/// structures that contain both differentiable vector properties and other
+/// stored properties that do not have a derivative, e.g.
+///
+/// ```swift
+/// struct Perceptron: @memberwise EuclideanDifferentiable {
+///     var weight: SIMD16<Float>
+///     var bias: Float
+///     @noDerivative var useBias: Bool
+/// }
+/// ```
+///
+/// - Note: Conform a type to `EuclideanDifferentiable` if it is differentiable
+///   only with respect to its vector space component and when its
+///   `TangentVector` is equal to its vector space component.
+public protocol EuclideanDifferentiable: Differentiable {
+  /// The differentiable vector component of `self`.
+  var vectorView: TangentVector { get set }
```

Comment:
I thought we were planning on renaming TangentVector to simply Tangent. Are we sure that vectorView is the right name here? (I think so, but just wanted to flag this. Better yet, could we mark this as unstable for now?)

Reply:
We were never planning on renaming `TangnetVector` to `Tangent`. There are multiple reasons `TangentVector` was named this way: 1. Swift type names describe instances, not spaces; 2. domain-specific type names should be Google-able: "tangent vector" is while "tangent" is not.

Reply:
To your question about the naming of `vectorView`, I am sure this is the right name after analyzing various options, bu like any names, this design is to be brought up in a design review. Marking this property as unable does not feel particularly useful since the entire feature is experimental.